syntax = "proto3";

package gis.protobuf;

option java_package = "com.github.tomi77.gis";
option java_multiple_files = true;

import "gis/protobuf/point.proto";

// type: Polygon
// coordinates: [[[x, y], [x, y]]]
message Polygon2D {
    repeated gis.protobuf.Point2D point = 1;
}

// type: Polygon
// coordinates: [[[x, y, z], [x, y, z]]]
message Polygon3D {
    repeated gis.protobuf.Point3D point = 1;
}

// type: MultiPolygin
// coordinates: [[[[x, y], [x, y]]], [[[x, y], [x, y]]]]
message MultiPolygon2D {
    repeated Polygon2D polygon = 1;
}

// type: MultiPolygin
// coordinates: [[[[x, y, z], [x, y, z]]], [[[x, y, z], [x, y, z]]]]
message MultiPolygon3D {
    repeated Polygon3D polygon = 1;
}
